src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -10,45 +10,397 @@
+    px5 = _mm_packus_epi32(px6, pZero);    /* pack pixels 8-15 as B09-12|X01-04 */ \
+    px2 = _mm_packus_epi16(px4, px5);    /* pack pixels 0-15 as [R09|R10|R11|R12|G09|G10|G11|G12|B09|B10|B11|B12|00|00|00|00] */ \
+    px4 = _mm_cvtps_epi32(p3R);    /* convert to int32 for R13-16 */ \
+    px5 = _mm_cvtps_epi32(p3G);    /* convert to int32 for G13-16 */ \
+    px6 = _mm_cvtps_epi32(p3B);    /* convert to int32 for B13-16 */ \
+    px4 = _mm_packus_epi32(px4, px5);    /* pack pixels 0-7 as R13-16|G13-16 */ \
+    px5 = _mm_packus_epi32(px6, pZero);    /* pack pixels 8-15 as B13-16|X01-04 */ \
+    px3 = _mm_packus_epi16(px4, px5);    /* pack pixels 0-15 as [R13|R14|R15|R16|G13|G14|G15|G16|B13|B14|B15|B16|00|00|00|00] */ \
+    px0 = _mm_shuffle_epi8(px0, mask);    /* shuffle to get [R01|G01|B01|R02|G02|B02|R03|G03|B03|R04|G04|B04|00|00|00|00] */ \
+    px1 = _mm_shuffle_epi8(px1, mask);    /* shuffle to get [R05|G05|B05|R06|G06|B06|R07|G07|B07|R08|G08|B08|00|00|00|00] */ \
+    px2 = _mm_shuffle_epi8(px2, mask);    /* shuffle to get [R09|G09|B09|R10|G10|B10|R11|G11|B11|R12|G12|B12|00|00|00|00] */ \
+    px3 = _mm_shuffle_epi8(px3, mask);    /* shuffle to get [R13|G13|B13|R14|G14|B14|R15|G15|B15|R16|G16|B16|00|00|00|00] */ \
+    _mm_storeu_si128((__m128i *)dstPtrTemp, px0);           /* store [R01|G01|B01|R02|G02|B02|R03|G03|B03|R04|G04|B04|00|00|00|00] */ \
+    _mm_storeu_si128((__m128i *)(dstPtrTemp + 12), px1);    /* store [R05|G05|B05|R06|G06|B06|R07|G07|B07|R08|G08|B08|00|00|00|00] */ \
+    _mm_storeu_si128((__m128i *)(dstPtrTemp + 24), px2);    /* store [R09|G09|B09|R10|G10|B10|R11|G11|B11|R12|G12|B12|00|00|00|00] */ \
+    _mm_storeu_si128((__m128i *)(dstPtrTemp + 36), px3);    /* store [R13|G13|B13|R14|G14|B14|R15|G15|B15|R16|G16|B16|00|00|00|00] */ \
+}
+
+#define RPP_LOAD16_U8_TO_F32(srcPtrTemp) \
+{ \
+    px0 =  _mm_loadu_si128((__m128i *)srcPtrTemp);    /* load pixels 0-15 */ \
+    px1 = _mm_unpackhi_epi8(px0, zero);    /* pixels 8-15 */ \
+    px0 = _mm_unpacklo_epi8(px0, zero);    /* pixels 0-7 */ \
+    p0 = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px0, zero));    /* pixels 0-3 */ \
+    p1 = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px0, zero));    /* pixels 4-7 */ \
+    p2 = _mm_cvtepi32_ps(_mm_unpacklo_epi16(px1, zero));    /* pixels 8-11 */ \
+    p3 = _mm_cvtepi32_ps(_mm_unpackhi_epi16(px1, zero));    /* pixels 12-15 */ \
+}
+
+#define RPP_STORE16_F32_TO_U8(dstPtrTemp) \
+{ \
+    px0 = _mm_cvtps_epi32(p0);    /* pixels 0-3 */ \
+    px1 = _mm_cvtps_epi32(p1);    /* pixels 4-7 */ \
+    px2 = _mm_cvtps_epi32(p2);    /* pixels 8-11 */ \
+    px3 = _mm_cvtps_epi32(p3);    /* pixels 12-15 */ \
+    px0 = _mm_packus_epi32(px0, px1);    /* pixels 0-7 */ \
+    px1 = _mm_packus_epi32(px2, px3);    /* pixels 8-15 */ \
+    px0 = _mm_packus_epi16(px0, px1);    /* pixels 0-15 */ \
+    _mm_storeu_si128((__m128i *)dstPtrTemp, px0);    /* store pixels 0-15 */ \
+}
+
+#define RPP_LOAD12_F32PKD3_TO_F32PLN3(srcPtrTemp) \
```

Comment:
it is not a good practice to use variables from calling routine inside a macro.

Reply:
Done